### PR TITLE
DFNS-2125 feat: add flutter sdk

### PR DESCRIPTION
**Pull request by gaj-dfns**

### CODE REVIEW

File: lib/passkeys_signer.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import 'dart:convert';
```
Comment (djin-dfns) [inline lib/passkeys_signer.dart:1]: can we separate the sdk and the example app?

Comment (gaj-dfns) [reply lib/passkeys_signer.dart:1]: Yes, I'm planning to do that and publish the lib in the next PR

File: lib/passkeys_signer.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        ),
+        pubKeyCredParams: List<PubKeyCredParamType>.from(
+          challenge.pubKeyCredParams.map(
+            (e) => PubKeyCredParamType(
+              type: e.type,
+              alg: e.alg,
+            ),
+          ),
+        ),
+        timeout: null,
```
Comment (djin-dfns) [inline lib/passkeys_signer.dart]: for react native, I made this [a configurable option](https://github.com/dfns/dfns-sdk-ts/blob/m/packages/sdk-react-native/index.ts#L40).

File: lib/passkeys_signer.dart [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            type: e.type,
+            id: e.id,
+            transports: [],
+          ),
+        )),
+        mediation: MediationType.Required,
+      ),
+    );
+
+    return UserActionAssertion(
```
Comment (djin-dfns) [inline lib/passkeys_signer.dart]: to be consistent, this should return just the `Fido2Assertion`.

**Review by djin-dfns — APPROVED** [review]

File: lib/passkeys_signer.dart (hunk 2 above)
Comment (djin-dfns) [reply lib/passkeys_signer.dart]: what I mean is make the class not static, and on construction, take in an option that configures the timeout.

**Pull request merged**